scenario_gym/metrics/trajectory.py

```diff
@@ -64,3 +68,1126 @@ def _step(self, state: State) -> None:
+                    self.safe_long_dist_brake_dict[state.t] = []
+                self.safe_long_dist_brake_dict[state.t].append(
+                    {
+                        "safe_longitudinal_distance_brake": {
+                            f"{entity}": safeLonDisBrake
+                        }
+                    }
+                )
+
+                if state.t not in self.long_dist_dict:
+                    self.long_dist_dict[state.t] = []
+                self.long_dist_dict[state.t].append(
+                    {"longitudinal_distance": {f"{entity}": long_dist}}
+                )
+
+                if state.t not in self.long_risk_dict:
+                    self.long_risk_dict[state.t] = []
+                self.long_risk_dict[state.t].append(
+                    {"longitudinal_risk": {f"{entity}": long_risk}}
+                )
+
+    def get_state(self) -> float:
+        """Return the current distance travelled."""
+        return (
+            self.safe_long_dist_dict,
+            self.safe_long_dist_brake_dict,
+            self.long_dist_dict,
+            self.long_risk_dict,
+        )
+
+
+class SafeLatDistance(Metric):
+    """
+    Determines safe latitudinal distance.
+
+    Speed and acceleration inputs must be for the lateral axis
+    responseTime: time it takes rear car to react and begin braking
+    speedLeft: current velocity of left car
+    speedRight: current velocity of right car
+    accMaxResp: max acceleration of both cars towards each other during response
+    accMinBrake: min braking of both cars
+    """
+
+    name = "safe_latitudinal_distance"
+
+    def _reset(self, state: State) -> None:
+        """Find the ego."""
+        self.ego = state.scenario.ego
+        self.safe_lat_dist_dict = {}
+        self.safe_lat_dist_brake_dict = {}
+        self.lat_dist_dict = {}
+        self.lat_risk_dict = {}
+        # Time it takes rear car to begin braking (longitudinal and lateral).
+        # https://trl.co.uk/sites/default/files/PPR313_new.pdf
+        # Unit: seconds (s)
+        self.responseTime = 1.5
+        # Max acceleration of both cars towards
+        # each other during response time (lateral).
+        # Unit: m/s^2
+        self.accMaxResp = 1
+        # Min braking of both cars (lateral).
+        # https://doi.org/10.1007/s12544-013-0120-2
+        # Unit: m/s^2
+        self.accMinBrake = 1.5
+        # Max braking capability of both cars (lateral).
+        # Replace for ACC_MIN_BRAKE when calculating 'safeDistanceBrake'.
+        # https://doi.org/10.1007/s12544-013-0120-2
+        # Unit: m/s^2
+        self.accMaxBrake = 4
+
+    def _get_lane_and_lane_center_index(
+        self, entity_pos, entity_road_network_information
+    ):
+        # Initialize variables to deduce the closest lane center coordinates
+        min_distance = float("inf")
+        final_lane_center_index = None
+        final_lane_index = None
+
+        if "Lane" not in entity_road_network_information[0]:
```

Comment:
Could `entity_road_network_information[0]` be a var? its getting called a couple of times